### DISCUSSION TITLE
Air Installer in Wine

### DISCUSSION THREAD

**Discussion by sk8nfool** [airsdk/Adobe-Runtime-Support#1788] [Q&A]
I tried to run the Air Installer withing WINE - the Window emulator for Linux. I'm currently using Ubuntu 18.04. It fails for some unknown reason with the error "This application requires a version of Adobe AIR which cannot be found. Please download the latest version of the runtime from http://www.adobe.com/go/getair, or contact the application author for an updated version." - obviously a bit dated.  So I went to https://services.harman.com/partners/adobe (cause the Adobe site said to) and basically wound up here. I'm not a windows person and need a bit of help getting started. Any suggestions?

**Comment by ajwfrost** [comment]
That message comes up if it can't open the AIR runtime library... I've just checked (with a little utility we wrote for checking compatibility) and it seems that WINE has a kernel32 implementation that's missing support for "RaiseFailFastException"
https://docs.microsoft.com/en-us/windows/win32/api/errhandlingapi/nf-errhandlingapi-raisefailfastexception

We've just checked and this use of the "RaiseFailFastException" is fairly new and for Win7 and up (but based on build flags rather than runtime capabilities) - it should be relatively trivial to change this so that we look for the function dynamically and if it's not there, we just crash the app the old-fashioned way... This "fail fast" operation is only for if something goes fundamentally wrong and the whole thing would become unstable/untrusted, it's just like an 'abort()' call...

So basically .. in a few weeks, our latest installer should then allow you to load it in WINE... (assuming WINE supports everything else we need...)

thanks

**Comment by HardCode4All** [comment]
It's three years later now, and I am trying to run the latest AIR runtime - version 51.2.1.5 in Wine, but it just does nothing at all. I need AIR for Boss Tone Studio for my guitar amp, so it's needed pretty urgently.

Any idea on how to proceed here?

**Reply by ajwfrost** [reply]
Hi
Did any of the previous installers work with Wine? I just checked and the suggestion above for that `RaiseFailFastException` was integrated, so this is obviously something else going wrong. I don't suppose Wine gives any outputs/logs for why it fails to run?
thanks

**Reply by HardCode4All** [reply]
I am pretty new to Linux since ~6 weeks, so I have no experience with Wine or AIR so far. Coming from Windows, I didn't even realize that Boss Tone Studio depends on AIR, it just worked somehow :)

Never thought of installing an older version and yes, version 33.1.1.744 did the trick! Now I'm having trouble with the driver for the Katana Amp, but that's a different story. Thanks for the quick help!

Just for the record, here's the error I get when I run the AIR installer v51.2.1.5 on Winetricks 20250102-1 with Wine 10.10-1 on CachyOS:

WINEPREFIX="/apps/wineprefixes/BossKatana100/" wine ~/Downloads/progz-WIN/Boss\ Katana/BossToneStudio_katana_w400/AdobeAIR_51.2.1.5.exe

Unhandled exception: stack overflow in wow64 32-bit code (0x7252b1c3).
Register dump:
CS:0023 SS:002b DS:002b ES:002b FS:0063 GS:002b
EIP:7252b1c3 ESP:00034d30 EBP:00034d38 EFLAGS:00010206(  R- --  I   - -P- )
EAX:00031000 EBX:00000000 ECX:00031ca4 EDX:00826070
ESI:0420c6f8 EDI:0420cb48
Stack dump:
0x00034d30:  0420c6f8 7261b3ef 00034f64 7261b3a1
0x00034d40:  00034d58 71ec2fc0 726533c0 0420c718
0x00034d50:  041ff04c 041fdc00 00620061 0075006f
0x00034d60:  003a0074 006c0062 006e0061 0000006b
0x00034d70:  00000000 00000000 00000000 00000000
0x00034d80:  00000000 00000000 00000000 00000000
Backtrace:
0 0x7252b1c3 in ieframe (+0xb1c3) (0x00034d38)
1 0x7261b3a1 in ieframe (+0xfb3a1) (0x00034f64)
2 0x7266d319 in ieframe (+0x14d319) (0x00036204)
3 0x72616324 in ieframe (+0xf6324) (0x000364f4)
4 0x7262c132 in ieframe (+0x10c132) (0x00036520)
5 0x725498f7 in ieframe (+0x298f7) (0x0003654c)
6 0x761c55f9 in adobe air (+0x3a55f9) (0x0003657c)
7 0x72670f1e in ieframe (+0x150f1e) (0x000365ac)
8 0x7266d3a1 in ieframe (+0x14d3a1) (0x0003785c)
9 0x72616324 in ieframe (+0xf6324) (0x00037b4c)
10 0x7262c132 in ieframe (+0x10c132) (0x00037b78)
11 0x725498f7 in ieframe (+0x298f7) (0x00037ba4)
12 0x761c55f9 in adobe air (+0x3a55f9) (0x00037bd4)
13 0x72670f1e in ieframe (+0x150f1e) (0x00037c04)
14 0x7266d3a1 in ieframe (+0x14d3a1) (0x00038eb4)
15 0x72616324 in ieframe (+0xf6324) (0x000391a4)
16 0x7262c132 in ieframe (+0x10c132) (0x000391d0)
17 0x725498f7 in ieframe (+0x298f7) (0x000391fc)
18 0x761c55f9 in adobe air (+0x3a55f9) (0x0003922c)
19 0x72670f1e in ieframe (+0x150f1e) (0x0003925c)

From the error I gathered it might have to do with Internet Explorer (because of ieframe), so I installed ie8 but that didn't help either. 

The exception also talks about "stack overflow in wow64 32-bit code" so I tried to create a new 32-bit Wineprefix, but that seems not supported by Wine any more? see: https://bbs.archlinux.org/viewtopic.php?id=306356

Nevemind. Thanks again for the support, I'll stick to version 33.1.1.744 for now.

**Reply by ajwfrost** [reply]
I don't know the specifics about this application but it sounds like they use the feature in AIR which requires a web browser runtime. The old versions (up to 33.1.1.744) contain that library and that's probably why it worked; the more recent ones redirect those calls to a platform-specific web browser control (and on Windows, the default is still the "MSHTML" engine that was used by Internet Explorer and is used by Edge in their "IE Mode". So that's probably the issue you're seeing...

Good to know it works though; I don't know whether you'd also have had some luck if you managed to get hold of Adobe's original AIR runtime for Linux, before they dropped support (a long long time ago..)

thanks
